test/unit/variable_unit_test.rb

```diff
@@ -147,4 +147,16 @@ def test_output_raw_source_of_variable
     var = Variable.new(%! name_of_variable | upcase !)
     assert_equal " name_of_variable | upcase ", var.raw
   end
+
+  def test_variable_lookup_interface
+    lookup = VariableLookup.new('a.b.c')
+    assert_equal 'a', lookup.name
+    assert_equal ['b', 'c'], lookup.lookups
+    assert_equal 0, lookup.command_flags
+
+    lookup = VariableLookup.new('a.first.size')
+    assert_equal 'a', lookup.name
+    assert_equal ['first', 'size'], lookup.lookups
+    assert_equal 0b11, lookup.command_flags
```

Comment:
Doesn't really need to be in this test, but it's a bitfield to indicate which lookups are to be treated as method calls, i.e. size, first, and last (`VariableLookup::COMMAND_METHODS`). In fact, this ivar probably doesn't need to be exposed at all, it's just used internally in `VariableLookup#evaluate`.